### PR TITLE
Prevent forms from being closed if there are unsaved changes

### PR DESCRIPTION
**Pull request by pa-lem**
* Prevent form to be closed if the form has been updated

image

### CODE REVIEW

**Comment by dgarros** [comment]
Could this one goes into `stable` to be included in 0.16.1 ?

File: changelog/4419.added.md [hunk 1]
```diff
@@ -0,0 +1 @@
+* Prevent form to be clsoed if the form has been updated
```
Comment (bilalabbad) [inline changelog/4419.added.md]: typo

File: frontend/app/src/components/display/slide-over.tsx [hunk 2]
```diff
@@ -1,11 +1,12 @@
 import { Dialog, Transition } from "@headlessui/react";
-import React, { Fragment, useRef } from "react";
+import React, { Fragment, useRef, useState } from "react";
 import { Badge } from "@/components/ui/badge";
 import { Icon } from "@iconify-icon/react";
 import { ObjectHelpButton } from "@/components/menu/object-help-button";
 import { useAtomValue } from "jotai/index";
 import { currentBranchAtom } from "@/state/atoms/branches.atom";
 import { IModelSchema } from "@/state/atoms/schema.atom";
+import ModalConfirm from "../modals/modal-confirm";
```
Comment (bilalabbad) [inline frontend/app/src/components/display/slide-over.tsx]: Why ModalConfirm instead of ModalDelete? The green button made me think twice. I feel like a red button is more explicit that's it destructive operation. WDYT?

File: changelog/4419.added.md [hunk 3]
```diff
@@ -0,0 +1 @@
+* Prevent form to be closed if the form has been updated
```
Comment (bilalabbad) [inline changelog/4419.added.md]: How about:
`Prevent the form from being closed if there are unsaved changes.`

File: frontend/app/src/components/filters/filters.tsx [hunk 4]
```diff
@@ -8,9 +8,9 @@ import { useState } from "react";
 import ObjectForm from "@/components/form/object-form";
 import usePagination from "@/hooks/usePagination";
 import { IModelSchema } from "@/state/atoms/schema.atom";
-import { getFiltersFromFormData } from "@/components/filters/utils/getFiltersFromFormData";
 import { FormFieldValue } from "@/components/form/type";
 import { getObjectFromFilters } from "@/components/filters/utils/getObjectFromFilters";
+import { getFiltersFromFormData } from "./utils/getFiltersFromFormData";
```
Comment (bilalabbad) [inline frontend/app/src/components/filters/filters.tsx]: it looks like an unnecessary changes right? Can we remove it to prevent added commit to history?

File: frontend/app/src/components/ui/form.tsx [hunk 5]
```diff
@@ -47,6 +59,10 @@ export const Form = React.forwardRef<FormRef, FormProps>(
               event.stopPropagation();
             }
 
+            if (slideOverContext?.setPreventClose) {
+              slideOverContext?.setPreventClose(false);
+            }
+
```
Comment (bilalabbad) [inline frontend/app/src/components/ui/form.tsx]: if we submit and the submit fails validation, we are able to close the drawer again with unsaved changes

**Review by bilalabbad — APPROVED** [review]

**Comment by pa-lem** [comment]
only nats is failing, will merge
